dashboard/route.js

```diff
@@ -36,17 +36,17 @@ module.exports = function(app, ini) {
 	app.get('/dashboard/activities', authController.validateSession, activitiesController.index);
 	app.get('/dashboard/activities/launch', authController.validateSession, activitiesController.fakeLaunch);
 	app.get('/dashboard/activities/launch/:jid', authController.validateSession, activitiesController.launch);
-	app.get('/dashboard/stats', authController.validateSession, statsController.index);
+	app.get('/dashboard/stats', authController.validateSession, authController.checkRole(statsController.index));
 	app.get('/dashboard/stats/graph', authController.validateSession, statsController.getGraph);
 	app.get('/dashboard/graph', authController.validateSession, graphController.getGraph);
 
 	// classrooms routes
 	app.get('/dashboard/classrooms', authController.validateSession, classroomsController.index);
-	app.get('/dashboard/classrooms/add', authController.validateSession, classroomsController.addClassroom);
-	app.post('/dashboard/classrooms/add', authController.validateSession, classroomsController.addClassroom);
+	app.get('/dashboard/classrooms/add', authController.validateSession, authController.checkRole(classroomsController.addClassroom));
```

Comment:
instead of this
```nodejs
app.get('/dashboard/classrooms/add', authController.validateSession, authController.checkRole(classroomsController.addClassroom));
```
you can do this
```nodejs
app.get('/dashboard/classrooms/add', authController.validateSession, authController.checkRole,  classroomsController.addClassroom);
```

Reply:
Actually, checkRole is a function which is used to switch between Views for different roles.
The function can be used as `checkRole(adminRoute, teacherRoute)`. If the role is `Admin` then the router directs it to the Admin Route else to the Teacher Route If the Teacher Route parameter is empty, it directs it to the 404 Page.